delivery_send_to_shipper_at_operation/models/stock_picking.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from lxml import etree
+
+from odoo import fields, models
+from odoo.osv import expression
+from odoo.tools.safe_eval import safe_eval
+
+from odoo.addons.base.models.ir_ui_view import (
+    transfer_modifiers_to_node,
+    transfer_node_to_modifiers,
+)
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    delivery_notification_sent = fields.Boolean(default=False)
+
+    def _send_confirmation_email(self):
+        for picking in self:
+            picking._handle_send_to_shipper_at_operation()
+            super(StockPicking, picking)._send_confirmation_email()
+
+    def _handle_send_to_shipper_at_operation(self):
```

Comment:
have you checked my comment regarding hooking to here in the old PR?

Reply:
Done, last fixup fixes this, now we add the delivery cost to the SO only on the ship validation like before.
This module will only send the delivery notice to the carrier, nothing more.